Add text about deployment assumptions for servers (#143)
ietf-wg-ohai/oblivious-http: draft-ietf-ohai-ohttp.md

provide confidentiality protection for the unprotected requests and responses,
plus protections for traffic analysis; see {{ta}}.

> Unsecured requests - such those with the "http" scheme as opposed to the "https"
> scheme - MUST NOT be used if the Oblivious Gateway and Target Resources are
> operated by different entities as that would expose both requests and response
> to modification or inspection by a network attacker.

Reviewer: ```suggestion
```
I don't think we should restrict OHTTP in this way. It's like saying clients MUST NOT use "http" in the first place (if they weren't using OHTTP). The protocol still has benefits for clients even in the event that the target resource is only accessible over HTTP.

Author: I think that this is important.  Do you have some scenario in mind where this would be a problem?  Because I can easily see how sending requests (and responses) without integrity or confidentiality would be bad, but I can't see how it might be could be useful.

Reviewer 2: @chris-wood I think this is OK to restrict in this way. Do we have any cases of unencrypted HTTP servers that need to be accessible over OHTTP?

In general, the gateway needs to be tightly associated with the target, not just some random server, since the HPKE key config effectively stands in for the TLS authentication and encryption that otherwise would occur to the target.

If the connection between the gateway and the target is something that traverses an open network, people could look at and modify the content. This might be OK if the client's only goal is the server not learning the client IP address, but the client would then need to be very very careful about what data this is. Writing the text to guide clients sounds hard, and doesn't seem to provide much benefit?

Reviewer: For what it's worth, I would be comfortable with a "SHOULD NOT," but I am not OK with "MUST NOT."

Reviewer: > In general, the gateway needs to be tightly associated with the target, not just some random server, since the HPKE key config effectively stands in for the TLS authentication and encryption that otherwise would occur to the target.

This is not the security model for OHTTP. The target can be _any_ type of resource for OHTTP to work as intended (hide the client IP address).

Reviewer: > If the connection between the gateway and the target is something that traverses an open network, people could look at and modify the content. This might be OK if the client's only goal is the server not learning the client IP address, but the client would then need to be very very careful about what data this is. Writing the text to guide clients sounds hard, and doesn't seem to provide much benefit?

I think this is why a SHOULD NOT works here. Clients that don't follow the guidance risk sending plaintext application data exposed to a network, which is no different from sending plaintext application data over a network if they were not using OHTTP, with one exception: use of OHTTP changes where the attacker has access to that plaintext data.

> Unsecured requests - such those with the "http" scheme as opposed to the "https"
> scheme - MUST NOT be used if the Oblivious Gateway and Target Resources are
> operated by different entities as that would expose both requests and response
> to modification or inspection by a network attacker.

Reviewer: ```suggestion
Nonsecure requests - such those with the "http" scheme as opposed to the "https"
scheme - SHOULD NOT be used if the Oblivious Gateway and Target Resources are
operated by different entities as that would expose both requests and response
to modification or inspection by a network attacker.
```

Author: I just realized that maybe OHTTP could be used for CRL fetching, which a MUST NOT would exclude.  I'd still prefer something stronger, but I can live with this.

Reviewer: Bingo!

Reviewer: Citing CRL and OCSP queries as examples where the SHOULD is not followed seems like a good thing to do.
